Libplanet/Net/Swarm.cs

```diff
@@ -1330,12 +1374,21 @@ CancellationToken cancellationToken
             try
             {
                 _logger.Debug($"Trying to DialAsync({peer.EndPoint})...");
-                await DialAsync(
+                Pong pong = await DialAsync(
                     ToNetMQAddress(peer),
                     dealer,
                     cancellationToken);
                 _logger.Debug($"DialAsync({peer.EndPoint}) is complete.");
 
+                if (pong.ProtocolVersion != _protocolVersion)
+                {
+                    dealer.Dispose();
+                    throw new DifferentProtocolVersionException(
+                        $"Peer protocol version({pong.ProtocolVersion}) is " +
```

Comment:
We could have rather a well-structured exception type.  I mean, how about having a separated properties like `DifferentProtocolVersionException.ExpectedVersion` and `DifferentProtocolVersionException.ActualVersion`?  This approach is a more programmable way.  See also `DifferentProtocolVersionException` as an example of this approach.